docs/source/nested.rst

```diff
@@ -10,213 +10,471 @@ Introduction
+the reason for these errors and suggestions for how to address them.
 
-The following functions are related to nested tensors:
+.. _unimplemented_op:
+
+Unimplemented ops
+^^^^^^^^^^^^^^^^^
+
+This error is becoming rarer as nested tensor op support grows, but it's still possible to hit it
+today given that there are a couple thousand ops within PyTorch.
+
+::
+
+    NotImplementedError: aten.view_as_real.default
+
+The error is straightforward; we haven't gotten around to adding op support for this particular op
+yet. If you'd like, you can `contribute <contributions>`__ an implementation yourself OR simply
+`request <https://github.com/pytorch/pytorch/issues/118107>`_ that we add support for this op
+in a future PyTorch release.
+
+.. _ragged_structure_incompatibility:
+
+Ragged structure incompatibility
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+::
+
+    RuntimeError: cannot call binary pointwise function add.Tensor with inputs of shapes (2, j2, 128) and (2, j3, 128)
+
+This error occurs when calling an op that operates over multiple NJTs with incompatible ragged
+structures. Currently, it is required that input NJTs have the exact same ``offsets`` constituent
+in order to have the same symbolic ragged structure symbol (e.g. ``j1``).
+
+As a workaround for this situation, it is possible to construct NJTs from the ``values`` and
+``offsets`` components directly. With both NJTs referencing the same ``offsets`` components, they
+are considered to have the same ragged structure and are thus compatible.
+
+>>> a = torch.randn(50, 128)
+>>> b = torch.randn(32, 128)
+>>> nt1 = torch.nested.nested_tensor([a, b], layout=torch.jagged, dtype=torch.float32)
+>>> nt2 = torch.nested.nested_tensor_from_jagged(values=torch.randn(82, 128), offsets=nt1.offsets())
+>>> nt3 = nt1 + nt2
+>>> nt3.shape
+torch.Size([2, j1, 128])
+
+Data dependent operatation within torch.compile
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+::
+
+    torch._dynamo.exc.Unsupported: data dependent operator: aten._local_scalar_dense.default; to enable, set torch._dynamo.config.capture_scalar_outputs = True
+
+This error occurs when calling an op that does data-dependent operation within torch.compile; this
+commonly occurs for ops that need to examine the values of the NJT's ``offsets`` to determine the
+output shape. For example:
+
+>>> a = torch.randn(50, 128)
+>>> b = torch.randn(32, 128)
+>>> nt = torch.nested.nested_tensor([a, b], layout=torch.jagged, dtype=torch.float32)
+>>> def f(nt): return nt.chunk(2, dim=0)[0]
+...
+>>> compiled_f = torch.compile(f, fullgraph=True)
+>>> output = compiled_f(nt)
+
+In this example, calling ``chunk()`` on the batch dimension of the NJT requires examination of the
+NJT's ``offsets`` data to delineate batch item boundaries within the packed ragged dimension. The
+workaround here is as suggested in the error message.
+
+>>> torch._dynamo.config.capture_scalar_outputs = True
```

Comment:
Should we also mention `capture_dynamic_output_shape_ops=True` here?

Reply:
yeah good point, probably should! added